functions/geometry_vi

```diff
@@ -0,0 +1,77 @@
+# geometry_vi - A geometry plugin to display the ZLE vi-mode state.
+
+geometry_vi() {
+    # Required for RPROMPT to render properly after submitting a command.
+    geometry::vi-get-mode
+}
+
+# Re-render the prompt.
+function geometry::vi-update-prompt() {
+    imode=$(geometry::vi-insert-mode)
+    nmode=$(geometry::vi-normal-mode)
+    if [[ $KEYMAP == vicmd ]]; then
+        PROMPT=${PROMPT//$imode/$nmode}
```

Comment:
I was trying to find a clever way to pass the prompt as an indirect variable through `${(P)varname}` but couldn't find a way to assign back, so for now this is the cleanest way to do it.